darts/ad/utils.py

```diff
@@ -422,105 +425,92 @@ def show_anomalies_from_scores(
-
-    if pred_scores is not None:
-        dict_input = {}
-
-        for idx, (score, w) in enumerate(zip(pred_scores, window)):
-            dict_input[idx] = {"series_score": score, "window": w, "name_id": idx}
-
-        for index, elem in enumerate(
-            sorted(dict_input.items(), key=lambda x: x[1]["window"])
-        ):
-            if index == 0:
-                current_window = elem[1]["window"]
-                index_ax = index_ax + 1
-
-            idx = elem[1]["name_id"]
-            w = elem[1]["window"]
-
-            if w != current_window:
-                current_window = w
-                index_ax = index_ax + 1
-
-            if metric is not None:
-                value = round(
-                    eval_metric_from_scores(
-                        anomalies=anomalies,
-                        pred_scores=pred_scores[idx],
-                        window=w,
-                        metric=metric,
-                    ),
-                    3,
-                )
-            else:
-                value = None
-
-            if names_of_scorers is not None:
-                label = names_of_scorers[idx] + [f" ({value})", ""][value is None]
-            else:
-                label = f"score_{str(idx)}" + [f" ({value})", ""][value is None]
-
-            _plot_series(
-                series=elem[1]["series_score"],
-                ax_id=axs[index_ax][0],
-                linewidth=0.5,
-                label_name=label,
-            )
-
-            axs[index_ax][0].legend(
-                loc="upper center", bbox_to_anchor=(0.5, 1.19), ncol=2
+        for i in range(series_width):
+            index_ax = i * nbr_plots
+            _plot_series_and_anomalies(
+                series=series[series.components[i]],
+                anomalies=anomalies[anomalies.components[i]]
+                if anomalies is not None
+                else None,
+                pred_series=pred_series[pred_series.components[i]]
+                if pred_series is not None
+                else None,
+                pred_scores=pred_scores,
+                window=window,
+                names_of_scorers=names_of_scorers,
+                metric=metric,
+                axs=axs,
+                index_ax=index_ax,
+                nbr_plots=nbr_plots,
             )
-            axs[index_ax][0].set_title(f"Window: {str(w)}", loc="left")
-            axs[index_ax][0].set_title("")
-            axs[index_ax][0].set_xlabel("")
 
-    if anomalies is not None:
-        _plot_series(
-            series=anomalies,
-            ax_id=axs[index_ax + 1][0],
-            linewidth=1,
-            label_name="anomalies",
-            color="red",
+        fig.suptitle(title)
+    else:
+        fig, axs = plt.subplots(
+            nbr_plots,
+            figsize=(8, 4 + 2 * (nbr_plots - 1)),
+            sharex=True,
+            gridspec_kw={"height_ratios": [2] + [1] * (nbr_plots - 1)},
+            squeeze=False,
```

Comment:
Let's avoid having this snippet twice and call it before treating the two different cases by including `multivariate_plot` in the logic;

```python
plots_per_ts = nbr_plots * series_width if multivariate_series else nbr_plots
fig, axs = plt.subplots(
            plots_per_ts,
            figsize=(8, 4 + 2 * (plots_per_ts - 1)),
            sharex=True,
            gridspec_kw={"height_ratios": [2] + [1] * (plots_per_ts - 1)},
            squeeze=False,
)
```